sqlite_query.c

```diff
@@ -104,24 +302,111 @@ sqlite_convert_to_pg(Oid pgtyp, int pgtypmod, sqlite3_stmt * stmt, int stmt_coli
 				 * "regular" process because its already implemented and
 				 * working properly.
 				 */
-				if (sqlite_value_affinity == SQLITE_INTEGER || sqlite_value_affinity == SQLITE_FLOAT)
-				{
-					double		value = sqlite3_column_double(stmt, stmt_colid);
-					Datum		d = DirectFunctionCall1(float8_timestamptz, Float8GetDatum((float8) value));
-
-					return (struct NullableDatum) { d, false};
-				}
-				else
+				switch (sqlite_value_affinity)
 				{
-					valstr = sqlite_text_value_to_pg_db_encoding(stmt, stmt_colid);
+					case SQLITE_INTEGER:
+					case SQLITE_FLOAT:
+						{
+							double		value = sqlite3_column_double(stmt, stmt_colid);
+							Datum		d = DirectFunctionCall1(float8_timestamptz, Float8GetDatum((float8) value));
```

Comment:
Should we separate Timestamp and TimestampTz ?

Reply:
I think yes, but this code repeats current implementation and don't change this architecture. `timestamp` is also _mixed affinity_ case.

Reply:
I understood this. Thank for explaining.

Reply:
Resolved.